src/main/java/eu/europeana/metis/sandbox/controller/DatasetController.java

```diff
@@ -38,69 +39,117 @@
 class DatasetController {
 
   private static final String MESSAGE_FOR_PROCESS_DATASET = ""
-          + "<span style=\"font-style: normal; font-size: 125%; font-weight: 750;\">"
-          + "202 Accepted</span>"
-          + " - The response body will contain an object of type"
-          + " <span style=\"font-style: normal; font-size: 125%; font-weight: 750;\">"
-          + DatasetIdDto.SWAGGER_MODEL_NAME + "</span>.";
+      + "<span style=\"font-style: normal; font-size: 125%; font-weight: 750;\">"
+      + "202 Accepted</span>"
+      + " - The response body will contain an object of type"
+      + " <span style=\"font-style: normal; font-size: 125%; font-weight: 750;\">"
+      + DatasetIdDto.SWAGGER_MODEL_NAME + "</span>.";
 
   private static final String MESSAGE_FOR_RETRIEVE_DATASET = ""
-          + "<span style=\"font-style: normal; font-size: 125%; font-weight: 750;\">"
-          + "200 OK</span>"
-          + " - The response body will contain an object of type"
-          + " <span style=\"font-style: normal; font-size: 125%; font-weight: 750;\">"
-          + ProgressInfoDto.PROGRESS_SWAGGER_MODEL_NAME + "</span>.";
+      + "<span style=\"font-style: normal; font-size: 125%; font-weight: 750;\">"
+      + "200 OK</span>"
+      + " - The response body will contain an object of type"
+      + " <span style=\"font-style: normal; font-size: 125%; font-weight: 750;\">"
+      + ProgressInfoDto.PROGRESS_SWAGGER_MODEL_NAME + "</span>.";
 
   private static final Pattern namePattern = Pattern.compile("[a-zA-Z0-9_-]+");
 
   @Value("${sandbox.dataset.max-size}")
   private int maxRecords;
 
-  private final ZipService zipService;
+  private final HarvestService harvestService;
   private final DatasetService datasetService;
   private final DatasetReportService reportService;
 
-  public DatasetController(ZipService zipService,
+
+  public DatasetController(HarvestService harvestService,
       DatasetService datasetService,
       DatasetReportService reportService) {
-    this.zipService = zipService;
+    this.harvestService = harvestService;
     this.datasetService = datasetService;
     this.reportService = reportService;
   }
 
-  @ApiOperation("Process the given dataset")
+  @ApiOperation("Process the given dataset by HTTP providing a file")
+  @ApiResponses({
+      @ApiResponse(code = 202, message = MESSAGE_FOR_PROCESS_DATASET, response = Object.class)
+  })
+  @PostMapping(value = "dataset/{name}/harvestByFile", produces = APPLICATION_JSON_VALUE)
+  @ResponseStatus(HttpStatus.ACCEPTED)
+  public DatasetIdDto harvestDatasetFromFile(
+      @ApiParam(value = "name of the dataset", required = true) @PathVariable(value = "name") String datasetName,
+      @ApiParam(value = "country of the dataset", required = true, defaultValue = "Netherlands") @RequestParam Country country,
+      @ApiParam(value = "language of the dataset", required = true, defaultValue = "nl") @RequestParam Language language,
```

Comment:
Don't forget to change the default value for Language

Reply:
Done, changed accordingly.